src/linker/implementation.py

```diff
@@ -50,7 +50,7 @@ def run(
             spark_master_url, job_id = build_cluster(
```

Comment:
Nit: for readability, it might be better to import `spark_utils` so you have 
```suggestion
            spark_master_url, job_id = spark_utils.build_cluster(
```
Otherwise, it's not as clear that the cluster in question is a spark cluster.